agent/consul/state/acl_events.go

```diff
@@ -0,0 +1,81 @@
+package state
+
+import (
+	"github.com/hashicorp/consul/agent/agentpb"
+	"github.com/hashicorp/consul/agent/structs"
+	memdb "github.com/hashicorp/go-memdb"
+)
+
+// ACLEventsFromChanges returns all the ACL token, policy or role events that
+// should be emitted given a set of changes to the state store.
+func (s *Store) ACLEventsFromChanges(tx *txnWrapper, changes memdb.Changes) ([]agentpb.Event, error) {
+
+	// Don't allocate yet since in majority of update transactions no ACL token
+	// will be changed.
+	var events []agentpb.Event
+
+	getObj := func(change memdb.Change) interface{} {
```

Comment:
These two local functions could be promoted to be first class regular functions instead of stack allocated closures.

Reply:
Do you think they should be methods on `memdb.Change`? I could see that. They are kinda simple though my intention was just to try to keep the repetition/extra nesting out of the bodies of the switch here locally as it's already a lot to process. I'd tend to generalise them only once there was at least one other place where having a method in place of a single If statement was a real readability win? Changing memdb seems cleanest but also a bit of extra effort ;)